diff -urN pcmcia.orig/cardmgr.c pcmcia/cardmgr.c
--- pcmcia.orig/cardmgr.c	Thu Jun 29 11:47:03 2000
+++ pcmcia/cardmgr.c	Thu Jun 29 11:47:41 2000
@@ -63,6 +63,8 @@
 
 /*====================================================================*/
 
+#define syslog my_syslog
+
 typedef struct socket_info_t {
     int			fd;
     int			state;
@@ -107,7 +109,7 @@
 #endif
 
 /* Default path for pid file */
-static char *pidfile = "/var/run/cardmgr.pid";
+char *pidfile = "/var/run/cardmgr.pid";
 
 #ifdef __linux__
 /* Default path for finding modules */
@@ -616,7 +618,6 @@
     char line[256];
     
     syslog(LOG_INFO, "executing: '%s'", cmd);
-    strcat(cmd, " 2>&1");
     f = popen(cmd, "r");
     while (fgets(line, 255, f)) {
 	line[strlen(line)-1] = '\0';
@@ -1300,7 +1301,7 @@
 
 /*====================================================================*/
 
-int main(int argc, char *argv[])
+int cardmgr_main(int argc, char *argv[])
 {
     int optch, errflg;
     int i, max_fd, ret, event, pass;
diff -urN pcmcia.orig/probe.c pcmcia/probe.c
--- pcmcia.orig/probe.c	Thu Jun 29 11:47:03 2000
+++ pcmcia/probe.c	Thu Jun 29 11:51:01 2000
@@ -428,7 +428,7 @@
 
 /*====================================================================*/
 
-int main(int argc, char *argv[])
+int probe_main(int argc, char *argv[])
 {
     int optch, errflg;
     extern char *optarg;
@@ -451,19 +451,19 @@
     }
     if (errflg || (optind < argc)) {
 	fprintf(stderr, "usage: %s [-t tcic_base] [-v] [-m]\n", argv[0]);
-	exit(EXIT_FAILURE);
+	return(-1);
     }
 
 #ifdef CONFIG_PCI
     if (pci_probe(verbose, module) == 0)
-	exit(EXIT_SUCCESS);
+	return(2);
 #endif
 #ifdef CONFIG_ISA
     if (i365_probe(verbose, module) == 0)
-	exit(EXIT_SUCCESS);
+	return(2);
     else if (tcic_probe(verbose, module, tcic_base) == 0)
-	exit(EXIT_SUCCESS);
+	return(1);
 #endif
-    exit(EXIT_FAILURE);
+    return(-1);
     return 0;
 }
